apps/src/applab/applab.js

```diff
@@ -1330,8 +1331,30 @@ Applab.onPropertyChange = function(element, name, value) {
     case 'text':
       $(element).text(value);
       break;
+    case 'textColor':
+      element.style.color = value;
+      break;
+    case 'backgroundColor':
+      element.style.backgroundColor = value;
+      break;
+    case 'fontSize':
+      element.style.fontSize = value + 'px';
+      break;
+    case 'image':
+      // For now, we stretch to image to fit the element
+      var width = Applab.getOuterWidth(element);
+      var height = Applab.getOuterHeight(element);
+      element.style.backgroundImage = 'url(' + value + ')';
+      element.style.backgroundSize = width + 'px ' + height + 'px';
+      break;
+    case 'hidden':
+      // Add a class that shows as 30% opacity in design mode, and invisible
+      // in code mode.
+      $(element).toggleClass('design-mode-hidden', value === true);
+      break;
     default:
-      throw "unknown property name " + name;
+      console.warn("unknown property name " + name);
```

Comment:
I believe that we shim it in dashboard, such that this should be fine, but you're right that we avoid this when possible. I didn't actually intend to commit this - it was just for debugging purposes as I was adding new fields. Will go back to using throw